src/lib/pages/backoffice/Actions/CheckOut/state/actions.js

```diff
@@ -0,0 +1,113 @@
+import { itemApi } from '@api/items';
+import { patronApi } from '@api/patrons';
+import {
+  addNotification,
+  sendErrorNotification,
+} from '@components/Notifications';
+import { invenioConfig } from '@config';
+import { goTo } from '@history';
+import { BackOfficeRoutes } from '@routes/urls';
+
+export const CLEAR_SEARCH = 'checkOutSearch/CLEAR_SEARCH';
+export const CLEAR_RESULTS = 'checkOutSearch/CLEAR_RESULTS';
+export const QUERY_STRING_UPDATE = 'checkOutSearch/QUERY_STRING_UPDATE';
+export const SEARCH_IS_LOADING = 'checkOutSearch/SEARCH_IS_LOADING';
+export const SEARCH_HAS_ERROR = 'checkOutSearch/SEARCH_HAS_ERROR';
+export const SEARCH_PATRON_SUCCESS = 'checkOutSearch/SEARCH_PATRON_SUCCESS  ';
+export const SEARCH_ITEM_SUCCESS = 'checkOutSearch/SEARCH_ITEM_SUCCESS  ';
+
+export const updateQueryString = queryString => {
+  return dispatch => {
+    dispatch({
+      type: QUERY_STRING_UPDATE,
+      payload: queryString,
+    });
+  };
+};
+
+export const clearSearch = () => {
+  return dispatch => {
+    dispatch({
+      type: CLEAR_SEARCH,
+    });
+  };
+};
+
+export const clearResults = () => {
+  return dispatch => {
+    dispatch({
+      type: CLEAR_RESULTS,
+    });
+  };
+};
+
+const searchPatrons = async (dispatch, term) => {
+  const response = await patronApi.list(
+    `${invenioConfig.PATRONS.scanField.field}:${term}`
```

Comment:
This is for all reviewers: here it looks to me that we search in a specific field only. Do we want this?
What about if I don't have the Patron ID and I am looking for her/him by name? Or email?

We currently have:
![New_Request_-_CERN_Document_Server](https://user-images.githubusercontent.com/2089455/88381305-75a16c80-cda6-11ea-88c6-cba47784a550.png)

Reply:
We were discussing this with @topless and I think we focused too much on our use case. It might be better to search on any field

Reply:
Or maybe, it would be nice to have the possibility to configure multiple fields, with a `OR` clause: `person_id:Nicola OR name:Nicola OR email:Nicola`.
In case of email or name, e.g. Nicola, you will probably have multiple results, but in case of inputting a person id, then we should have only one result.

Reply:
@ntarocco, @kprzerwa, my impression was that this workflow would be used in assistance with the scanner, thus the restriction to the specific field. Nevertheless I agree with @ntarocco latest suggestion for a patron `queryBuilder` where we extend the usage to all inputs.

Reply:
The latest commit introduced queryBuilder for patronApi, with wildcard support on every field. We can tune it to search for the exact matches or be more loose and search for props that contain the search value.